src/main/kotlin/no/nav/familie/ef/iverksett/brukernotifikasjon/BrukernotifikasjonKafkaProducer.kt

```diff
@@ -0,0 +1,85 @@
+
+    @Value("\${KAFKA_TOPIC_DITTNAV}")
+    private lateinit var topic: String
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+    private val secureLogger = LoggerFactory.getLogger("secureLogger")
+
+    fun sendBeskjedTilBruker(iverksett: IverksettData, behandlingId: UUID) {
+        val nokkel = lagNøkkel(iverksett.søker.personIdent, behandlingId)
+        val beskjed = lagBeskjed(iverksett)
+
+        secureLogger.info("Sender til Kafka topic: {}: {}", topic, beskjed)
+        runCatching {
+            val producerRecord = ProducerRecord(topic, nokkel, beskjed)
+            kafkaTemplate.send(producerRecord).get()
+        }.onFailure {
+            val errorMessage = "Kunne ikke sende brukernotifikasjon til topic: $topic. Se secure logs for mer informasjon."
+            logger.error(errorMessage)
+            secureLogger.error("Kunne ikke sende brukernotifikasjon til topic: {}", topic, it)
+            throw RuntimeException(errorMessage)
+        }
+    }
+
+    private fun lagNøkkel(fnr: String, behandlingId: UUID): NokkelInput =
+        NokkelInputBuilder()
+            .withAppnavn("familie-ef-iverksett")
+            .withNamespace("teamfamilie")
+            .withFodselsnummer(fnr)
+            .withGrupperingsId(UUID.randomUUID().toString()) // Setter random UUID uten å lagre fordi feltet skal fjernes
+            .withEventId(behandlingId.toString())
+            .build()
+
+    fun lagBeskjed(iverksett: IverksettData): BeskjedInput {
+        val builder = BeskjedInputBuilder()
+            .withSikkerhetsnivaa(4)
+            .withSynligFremTil(null)
+            .withTekst(genererGOmregningMelding(iverksett))
+            .withTidspunkt(LocalDateTime.now(ZoneOffset.UTC))
+
+        return builder.build()
+    }
+}
+
+fun genererGOmregningMelding(iverksett: IverksettData): String {
+    val melding = if (iverksett.vedtak.tilkjentYtelse?.andelerTilkjentYtelse?.any { it.inntekt > 0 } == true) {
+        gOmregningMedArbeidsinntektMelding(
+            iverksett.vedtak.tilkjentYtelse?.andelerTilkjentYtelse?.first { it.periode.inneholder(YearMonth.now().plusMonths(1)) && it.inntekt > 0 }?.inntekt
+                ?: iverksett.vedtak.tilkjentYtelse?.andelerTilkjentYtelse?.first { it.periode.inneholder(YearMonth.now()) && it.inntekt > 0 }?.inntekt
+                ?: iverksett.vedtak.tilkjentYtelse?.andelerTilkjentYtelse?.first { it.inntekt > 0 }?.inntekt ?: throw Exception("Skulle ha funnet inntekt i andel tilkjent ytelse for behandling ${iverksett.behandling.behandlingId}"),
+        )
+    } else {
+        G_OMREGNING_UTEN_ARBEDSINNTEKT_MELDING
+    }
+
+    return melding
+}
+
+fun gOmregningMedArbeidsinntektMelding(inntekt: Int) = """
+    Fra ${nyesteGrunnbeløp.periode.fomDato.norskFormat()} har folketrygdens grunnbeløp økt til ${nyesteGrunnbeløp.grunnbeløp} kroner og din forventede inntekt er oppjustert til $inntekt kroner. Overgangsstønaden din er derfor endret.
```

Comment:
Kan vi risikere noe "tull" rundt mismatch i "g-verdi" mellom ef-sak og ef-iverksett her? 🤔 
Eller må vi bare huske å oppdatere "samtidig" på begge steder hvert år?

Reply:
Beste løsning vil nok være å sende inn g-omregningsdata fra ef-sak. Jeg foreslår vi lager en oppgave for dette, men beholder denne i år?